src/Middleware/Localization/src/RequestLocalizationMiddleware.cs

```diff
@@ -148,6 +148,30 @@ private static void SetCurrentThreadCulture(RequestCulture requestCulture)
         return null;
     }
 
+    private static CultureInfo? GetCultureInfo(
+        StringSegment cultureName,
+        IList<CultureInfo>? supportedCultures,
+        bool fallbackToParentCultures,
+        int currentDepth)
+    {
+        var culture = GetCultureInfo(cultureName, supportedCultures);
+
+        if (culture == null && fallbackToParentCultures && currentDepth < MaxCultureFallbackDepth)
+        {
+            try
+            {
+                culture = CultureInfo.GetCultureInfo(cultureName.ToString());
```

Comment:
It is better to short circuit exiting if the culture is invariant. I mean when `cultureName.ToString()` is empty string.

Reply:
@hishamco Can you please make this change? Other than that, this PR should be good to go.

Reply:
See linked PR which includes @tarekgh's feedback. Once we are comfortable that my new PR is good we can probably just close this one down.

Thanks for submitting the code @hishamco - I basically just lifted your change and added a unit test.

Reply:
The root of all cultures is the Invariant culture which has the name as empty string. If you encounter it that means you reached the super root. 

> if the string is empty, should we return CultureInfo.InvariantCulture (open question).

Is the Invariant culture one of the supported cultures in the list `supportedCultures`? if yes, then the answer of your question is yes too.

Reply:
@hishamco I've had a closer look at the code and I think returning null when cultureName is an empty string is the desirable approach. If we don't do this we needlessly spin down the max-depth when traversing the parent property.